src/db.c

```diff
@@ -248,8 +248,9 @@ void dbOverwrite(redisDb *db, robj *key, robj *val) {
  * All the new keys in the database should be created via this interface.
  * The client 'c' argument may be set to NULL if the operation is performed
  * in a context where there is no clear client performing the operation. */
-void genericSetKey(client *c, redisDb *db, robj *key, robj *val, int keepttl, int signal) {
-    if (lookupKeyWrite(db,key) == NULL) {
+void genericSetKeyLookup(client *c, redisDb *db, robj *key, robj *val,int keepttl,
+                         int signal, robj* keyFound) {
```

Comment:
I don't like the name ("Lookup" suffix), we should be more explicit as to what it does or what it expects.
how about:
```suggestion
#define SETKEY_KEEPTTL 1
#define SETKEY_SIGNAL 2
#define SETKEY_ALREADY_EXIST 4
#define SETKEY_DOESNT_EXIST 8
void setKey(client *c, redisDb *db, robj *key, robj *val, int flags) {
```

Then, if the neither of the last two flags is passed, it performs an internal lookup, and if one of them does exist, it skips it.

Reply:
@madolson what do you think about it ?

Reply:
ohh, i now see that `setKey` is taken, the "generic" is the advanced method.
i see there are actually just 4 calls to the advanced "genericSetKey"
maybe we should just refactor it and modify these 4 places rather than have another wrapper?

Reply:
I mentioned in one of the comments that flags also seemed reasonable, I don't feel strongly one way or the other.

Reply:
i think the name `genericSetKeyLookup` is odd.
and also the new argument isn't documented in the top comment.

so i was thinking that maybe we wanna have just one "generic" function (one that exposes all the various flags / booleans) rather than have two of them, each one adding more and more flags.
if we keep only one "generic" function and add a 3rd boolean, we'll have to go modify all the callers today, and then when we one day add the 4th boolean, we'll have to do that again.
so i think we better convert all these booleans to flags today, have only one "genetic" function, and modify all the callers.